app/services/profile_fields/remove.rb

```diff
@@ -16,8 +16,9 @@ def initialize(id)
     def call
       @profile_field = ProfileField.find(@id)
       if @profile_field.destroy
-        accessor = profile_field.attribute_name.to_s
-        Profile.undef_method(accessor) if accessor.in?(Profile.attributes)
+        accessor = @profile_field.attribute_name.to_sym
+        Profile.undef_method(accessor) if accessor.in?(Profile.stored_attributes[:data])
```

Comment:
Going through `stored_attributes[:data]` is not ideal from an encapsulation perspective but it's the most robust way I found to only undefine exactly what we want, i.e. a dynamically added `store_attribute`.